src/bin/cargo/main.rs

```diff
@@ -28,6 +29,13 @@ fn main() {
         }
     };
 
+    let nightly_features_allowed = matches!(&*features::channel(), "nightly" | "dev");
+    if nightly_features_allowed {
+        clap_complete::CompleteEnv::with_factory(|| cli::cli(&mut gctx))
+            .var("CARGO_COMPLETE")
```

Comment:
We need this documented in https://doc.rust-lang.org/cargo/reference/unstable.html